src/test_application/activescenes/scene_vehicles.cpp

```diff
@@ -276,79 +407,85 @@ Session setup_vehicle_spawn_vb(
     OSP_SESSION_UNPACK_DATA(parts,          TESTAPP_PARTS);
     OSP_SESSION_UNPACK_TAGS(parts,          TESTAPP_PARTS);
+    OSP_SESSION_UNPACK_DATA(signalsFloat,   TESTAPP_SIGNALS_FLOAT);
+    OSP_SESSION_UNPACK_TAGS(signalsFloat,   TESTAPP_SIGNALS_FLOAT);
 
     Session vehicleSpawnVB;
     OSP_SESSION_ACQUIRE_DATA(vehicleSpawnVB, topData, TESTAPP_VEHICLE_SPAWN_VB);
     OSP_SESSION_ACQUIRE_TAGS(vehicleSpawnVB, rTags,   TESTAPP_VEHICLE_SPAWN_VB);
 
-    rBuilder.tag(tgVBSpBasicInReq)      .depend_on({tgVBSpBasicInMod});
-    rBuilder.tag(tgVBPartReq)           .depend_on({tgVBPartMod});
-    rBuilder.tag(tgVBWeldReq)           .depend_on({tgVBWeldMod});
-    rBuilder.tag(tgVBMachReq)           .depend_on({tgVBMachMod});
+    rBuilder.tag(tgVbSpBasicInReq)      .depend_on({tgVbSpBasicInMod});
+    rBuilder.tag(tgVbPartReq)           .depend_on({tgVbPartMod});
+    rBuilder.tag(tgVbWeldReq)           .depend_on({tgVbWeldMod});
+    rBuilder.tag(tgVbMachReq)           .depend_on({tgVbMachMod});
 
     top_emplace< ACtxVehicleSpawnVB >(topData, idVehicleSpawnVB);
 
-    vehicleSpawnVB.task() = rBuilder.task().assign({tgSceneEvt, tgVhSpBasicInReq, tgVBSpBasicInReq, tgEntNew, tgPartMod, tgWeldMod, tgVhSpPartMod, tgVhSpWeldMod, tgVBPartMod, tgVBWeldMod}).data(
-            "Create parts for vehicle using VehicleData",
+    vehicleSpawnVB.task() = rBuilder.task().assign({tgSceneEvt, tgVsBasicInReq, tgVbSpBasicInReq, tgEntNew, tgPartMod, tgWeldMod, tgVsPartMod, tgVsWeldMod, tgVbPartMod, tgVbWeldMod}).data(
+            "Create part IDs for vehicles from VehicleData",
             TopDataIds_t{                  idVehicleSpawn,                    idVehicleSpawnVB,           idScnParts},
             wrap_args([] (ACtxVehicleSpawn& rVehicleSpawn, ACtxVehicleSpawnVB& rVehicleSpawnVB, ACtxParts& rScnParts) noexcept
     {
         std::size_t const newVehicleCount = rVehicleSpawn.new_vehicle_count();
+        ACtxVehicleSpawnVB &rVSVB = rVehicleSpawnVB;
+
         if (newVehicleCount == 0)
         {
             return;
         }
 
-        rVehicleSpawnVB.m_remapPartOffsets  .resize(newVehicleCount);
-        rVehicleSpawnVB.m_remapWeldOffsets  .resize(newVehicleCount);
+        rVSVB.m_remapPartOffsets            .resize(newVehicleCount);
+        rVSVB.m_remapWeldOffsets            .resize(newVehicleCount);
         rVehicleSpawn.m_newVhPartOffsets    .resize(newVehicleCount);
         rVehicleSpawn.m_newVhWeldOffsets    .resize(newVehicleCount);
 
-        // Count total parts and welds, and calculate offsets for remap vectors
+        // Count total parts and welds, and calculate offsets for remaps
 
         std::size_t partTotal       = 0;
         std::size_t remapPartTotal  = 0;
 
         std::size_t weldTotal       = 0;
         std::size_t remapWeldTotal  = 0;
 
+        for (NewVehicleId vhId = 0; vhId < newVehicleCount; ++vhId)
         {
-            auto itRemapPartOffsetsOut  = std::begin(rVehicleSpawnVB.m_remapPartOffsets);
```

Comment:
Replaced quite a few iterators to use index-access instead. It became extremely confusing, noisy, and generally poor for my sanity when it's impossible to tell which containers the iterators are writing to.